apollo-router-core/src/response.rs

```diff
@@ -56,12 +70,79 @@ impl Response {
     pub fn append_errors(&mut self, errors: &mut Vec<Error>) {
         self.errors.append(errors)
     }
+
+    pub fn from_bytes(service_name: &str, b: Bytes) -> Result<Response, FetchError> {
+        let mut value =
+            Value::from_bytes(b).map_err(|error| FetchError::SubrequestMalformedResponse {
+                service: service_name.to_string(),
+                reason: error.to_string(),
+            })?;
+
+        let (data, errors, extensions) = match &mut value {
+            Value::Object(object) => (
+                object.remove("data").unwrap_or_default(),
+                match object.remove("errors") {
+                    Some(Value::Array(v)) => {
+                        let res: Result<Vec<Error>, FetchError> = v
+                            .into_iter()
+                            .map(|v| Error::from_value(service_name, v))
+                            .collect();
+                        res?
```

Comment:
:yarn: same but more concise 
```suggestion
                        v
                            .into_iter()
                            .map(|v| Error::from_value(service_name, v))
                            .collect::<Result<Vec<Error>, FetchError>>()?
```